contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -0,0 +1,368 @@
+import {AccessControlEnumerable} from "./utils/access/AccessControlEnumerable.sol";
+
+import {UnstructuredStorage} from "./lib/UnstructuredStorage.sol";
+import {PausableUntil} from "./utils/PausableUntil.sol";
+
+import {Versioned} from "./utils/Versioned.sol";
+
+/// @notice Interface defining a Lido liquid staking pool
+/// @dev see also [Lido liquid staking pool core contract](https://docs.lido.fi/contracts/lido)
+interface IStETH is IERC20, IERC20Permit {
+    function getSharesByPooledEth(uint256 _pooledEthAmount) external view returns (uint256);
+}
+
+/// @notice Interface defining a Lido liquid staking pool wrapper
+/// @dev see WstETH.sol for full docs
+interface IWstETH is IERC20, IERC20Permit {
+    function unwrap(uint256 _wstETHAmount) external returns (uint256);
+    function getStETHByWstETH(uint256 _wstETHAmount) external view returns (uint256);
+    function stETH() external view returns (IStETH);
+}
+
+/// @title A contract for handling stETH withdrawal request queue within the Lido protocol
+/// @author folkyatina
+abstract contract WithdrawalQueue is AccessControlEnumerable, PausableUntil, WithdrawalQueueBase, Versioned {
+    using SafeCast for uint256;
+    using SafeERC20 for IWstETH;
+    using SafeERC20 for IStETH;
+    using UnstructuredStorage for bytes32;
+
+    /// Bunker mode activation timestamp
+    bytes32 internal constant BUNKER_MODE_SINCE_TIMESTAMP_POSITION =
+        keccak256("lido.WithdrawalQueue.bunkerModeSinceTimestamp");
+    /// Special value for timestamp when bunker mode is inactive (i.e., protocol in turbo mode)
+    uint256 public constant BUNKER_MODE_DISABLED_TIMESTAMP = type(uint256).max;
+
+    // ACL
+    bytes32 public constant PAUSE_ROLE = keccak256("PAUSE_ROLE");
+    bytes32 public constant RESUME_ROLE = keccak256("RESUME_ROLE");
+    bytes32 public constant FINALIZE_ROLE = keccak256("FINALIZE_ROLE");
+    bytes32 public constant BUNKER_MODE_REPORT_ROLE = keccak256("BUNKER_MODE_REPORT_ROLE");
+
+    /// @notice minimal possible sum that is possible to withdraw
+    uint256 public constant MIN_STETH_WITHDRAWAL_AMOUNT = 100;
+
+    /// @notice maximum possible sum that is possible to withdraw by a single request
+    /// Prevents accumulating too much funds per single request fulfillment in the future.
+    /// @dev To withdraw larger amounts, it's recommended to split it to several requests
+    uint256 public constant MAX_STETH_WITHDRAWAL_AMOUNT = 1000 * 1e18;
+
+    /// @notice Lido stETH token address to be set upon construction
+    IStETH public immutable STETH;
+    /// @notice Lido wstETH token address to be set upon construction
+    IWstETH public immutable WSTETH;
+
+    /// @notice Emitted when the contract initialized
+    /// @param _admin provided admin address
+    /// @param _caller initialization `msg.sender`
+    event InitializedV1(address _admin, address _pauser, address _resumer, address _finalizer, address _caller);
```

Comment:
Let's remove `_caller`at all and add `_bunkerTriggerer`
`event InitializedV1(address _admin, address _pauser, address _resumer, address _finalizer, address _bunkerTriggerer);`

Reply:
docs